src/Timer/Timer.h

```diff
@@ -0,0 +1,188 @@
+/**@file
+ * This file is part of the Leon3 BSP for the Test Environment.
+ *
+ * @copyright 2022 N7 Space Sp. z o.o.
+ *
+ * Leon3 BSP for the Test Environment is free software: you can redistribute 
+ * it and/or modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * Leon3 BSP for the Test Environment is distributed in the hope
+ * that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Leon3 BSP for the Test Environment. If not,
+ * see <http://www.gnu.org/licenses/>.
+ */
+
+/// \brief Timer hardware driver function prototypes and datatypes.
+
+/**
+ * @defgroup Timer Timer
+ * @ingroup Bsp
+ * @{
+ */
+
+#ifndef BSP_TIMER_H
+#define BSP_TIMER_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "TimerTypedefs.h"
+
+/// \brief Default interrupt handler with no action when the interrupt is called.
+extern Timer_InterruptHandler defaultInterruptHandler;
+
+/// \brief Configures an Timer base configuration flags.
+/// \param [in] baseConfigurationRegister Base configuration register for a specific timer group.
+void Timer_baseInit(volatile uint32_t *const baseConfigurationRegister);
+
+/// \brief Configures an Timer device based on a configuration descriptor.
+/// \param [in] timer Timer device descriptor.
+/// \param [in] config A configuration descriptor.
+void Timer_setConfig(volatile uint32_t *const timerControlRegister, volatile uint32_t *const timerReloadRegister, const Timer_Config *const config);
```

Comment:
I would also rename this to setConfigRegisters; I assume that the user is meant to use Timer_Apbctrl1_setConfig for setting the configuration.

Reply:
ack